docs/examples/arrow_example.ipynb

```diff
@@ -583,11 +841,227 @@
+    "\n",
+    "    This function is similar to the arrow_to_numpy function, but also supports asymmetric data.\"\"\"\n",
+    "    result = {}\n",
+    "    result_dtype = power_grid_meta_data[dataset_type][component_type].dtype\n",
+    "\n",
+    "    for name in result_dtype.names:\n",
+    "        dtype = result_dtype[name]\n",
+    "        if len(dtype.shape) == 0:\n",
+    "            # simple or symmetric data type\n",
+    "            if name in data.column_names:\n",
+    "                column_data = data.column(name).to_numpy()\n",
+    "                result[name] = column_data.astype(result_dtype[name])\n",
+    "        else:\n",
+    "            # asymmetric data type\n",
+    "            for phase_index, phase in enumerate(phases_suffix):\n",
+    "                phase_name = f\"{name}_{phase}\"\n",
+    "                if phase_name not in data.column_names:\n",
+    "                    continue\n",
+    "\n",
+    "                column_data = data.column(phase_name).to_numpy()\n",
+    "                if name not in result:\n",
+    "                    result[name] = np.empty(shape=len(column_data), dtype=result_dtype[name])\n",
+    "                result[name][:, phase_index] = column_data\n",
+    "    return result\n",
+    "\n",
+    "\n",
+    "asym_load_input_columnar = arrow_to_numpy_asym_columnar(asym_loads, DatasetType.input, ComponentType.asym_load)\n",
+    "\n",
+    "asym_load_input_columnar"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Combined representation of 3 phase values\n",
+    "\n",
+    "We start from complete 3 phases as a fixed size list array"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "asym_load: {'names': ['id', 'node', 'status', 'type', 'p_specified', 'q_specified'], 'formats': ['<i4', '<i4', 'i1', 'i1', ('<f8', (3,)), ('<f8', (3,))], 'offsets': [0, 4, 8, 9, 16, 40], 'itemsize': 64, 'aligned': True}\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "pyarrow.RecordBatch\n",
+       "id: int32\n",
+       "node: int32\n",
+       "status: int8\n",
+       "type: int8\n",
+       "p_specified: fixed_size_list<item: double>[3]\n",
+       "  child 0, item: double\n",
+       "q_specified: fixed_size_list<item: double>[3]\n",
+       "  child 0, item: double\n",
+       "----\n",
+       "id: [7,8]\n",
+       "node: [2,3]\n",
+       "status: [1,1]\n",
+       "type: [0,0]\n",
+       "p_specified: [[1,0.01,0.011],[2,2.5,450]]\n",
+       "q_specified: [[0.5,1500,0.1],[1.5,2.5,1500]]"
+      ]
+     },
+     "execution_count": 27,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# TODO Decisions: Asymmetric attributes should be fixed list array or individual phases or both?\n",
```

Comment:
since we only support 3phase asym attributes, let's keep it as is. the 3-dimensional struct for asymmetric is - i believe - the best we can do. or what exactly do you mean?

Reply:
From all conversations I can definitely confirm that having zero copy conversion is the main goal here. It was not possible via "p_specified_a", "p_specified_b", "p_specified_c". A Fixed list array can do it with few operations. Hence I shall go ahead and modify that too. 

It shall be clear when the PR is in review.

Reply:
Arrow doesn't really support multi-dimensional columns (AFAIK) so zero-copy can't be avoided here, most likely. you can investigate whether it is possible, but if not, don't spend too much time on it. other people with more knowledge of Arrow may optimize this.